collect_events.py

```diff
@@ -55,31 +102,58 @@ def save_events(events: list[dict]) -> None:
     with open(LOG_FILE, "w", encoding="utf-8") as file:
         json.dump(events, file, indent=2)
 
+    logging.info(f"Events saved to {LOG_FILE}")
+
 
-def get_last_event_time(filename: str) -> str | None:
+def read_state() -> dict | None:
     """
     Get the time of the last event downloaded from the log file.
 
-    :param filename: Path to the log file.
     :return: Time of the last event, or None if the file does not exist.
     """
-    if not os.path.exists(filename):
-        return None
-
-    with open(filename, "r", encoding="utf-8") as file:
-        events = json.load(file)
-        return events[-1].get("published")
-
-
-def main():
-    last_event_time = get_last_event_time(LOG_FILE)
-
-    events = fetch_events(last_event_time)
-    print(f"Found {len(events)} new events")
+    if not os.path.exists(STATE_FILE):
+        return {}
+
+    with open(STATE_FILE, "r", encoding="utf-8") as file:

```

Comment:
nit: if the file it's empty, this will throw an exception. Maybe we can have a try except that also returns a {} but logs a warning?

Reply:
I think that's an edge case that we can ignore for simplicity.